src/olympia/devhub/tasks.py

```diff
@@ -281,6 +292,21 @@ def annotate_upgrade_warning(results, file_, addon, version_string):
             'message': msg,
             'description': [],
             'compatibility_type': None})
+        results['warnings'] += 1
```

Comment:
is `results['warnings']` guaranteed to be defined, even when it's `0`?  (`+=` needs the value to be already defined, right?)